src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const StateCopy = { ...state };
+  const res = [];
+
+  for (const key of actions) {
+    if (key.type === 'addProperties') {
+      for (const i in key.extraData) {
+        StateCopy[i] = key.extraData[i];
+      }
+      res.push({ ...StateCopy });
+    }
+
+    if (key.type === 'removeProperties') {
+      for (const i of key.keysToRemove) {
+        delete StateCopy[i];
+      }
+      res.push({ ...StateCopy });
+    }
+
+    if (key.type === 'clear') {
+      for (const i in StateCopy) {
```

Comment:
Using `for...in` to iterate over `StateCopy` is appropriate here since `StateCopy` is an object. However, ensure that there are no inherited properties that could affect the iteration.